app/native/src/components/relay/PrivateApiRenderer.js

```diff
@@ -1,53 +1,31 @@
 // @flow
 
 import * as React from 'react';
-import { QueryRenderer } from 'react-relay';
 import { connect } from 'react-redux';
-import { FullPageLoading } from '@kiwicom/native-common';
 
-import createEnvironment from '../../services/relay/Environment';
-import GeneralError from '../../components/errors/GeneralError';
 import Login from '../../components/authentication/Login';
+import QueryRenderer from './QueryRenderer';
 
 import type { ReduxState } from '../../types/Redux';
+import type { Props as PublicProps } from './PublicApiRenderer';
 
 type Props = {|
-  query: string,
-  render: ({ error: Object, props: Object }) => React.Node,
-  variables?: Object,
-  cacheConfig?: {|
-    force: boolean,
-  |},
+  ...PublicProps,
```

Comment:
I am amazed this works. Usually, it's not possible to spread exact types. Maybe something else is wrong?